pyomo/contrib/doe/utils.py

```diff
@@ -100,3 +100,43 @@ def rescale_FIM(FIM, param_vals):
 #             pass  # ToDo: Write error for suffix keys that aren't ParamData or VarData
 #
 #     return param_list
+
+
+def generate_snake_zigzag_pattern(*lists):
```

Comment:
Let's come up with a better name. I do not think of "snake" as a technical term.